### PR TITLE
capacity provider support

### PR DESCRIPTION
**Pull request by darrenweiner**
I'm not a java developer! But I really needed this feature in the most up to date jenkinsci.
full credit goes to https://github.com/owengo/amazon-ecs-plugin/, I just reconciled those with the most recent jenkinsci repo.
A bit hacky...but seems to work great.  Hope someone can take this and dial it in and merge it up.

### CODE REVIEW

**Comment by weirdninja** [comment]
@webratz  I have been leveraging this approach for quite a while now and it works alright. Assuming ASGs and capacity providers are properly configured, that is. Is there anything else needed to get this merged?

**Comment by webratz** [comment]
please see https://github.com/jenkinsci/amazon-ecs-plugin/issues/200

**Comment by charlesrg** [comment]
Please can this pull request be accepted ?

**Comment by jsbarber-emis** [comment]
Been testing this without issue. Seems like a simple, low risk solution to a significant problem (when coupled with correct ASG and capacity-provider setup). Be really helpful to get this merged if at all possible, although I appreciate the plugin is up for adoption.

File: src/main/java/com/cloudbees/jenkins/plugins/amazonecs/ECSService.java [hunk 1]
```diff
@@ -420,6 +420,10 @@ RunTaskResult runEcsTask(final ECSSlave agent, final ECSTaskTemplate template, S
             req.withPlatformVersion(template.getPlatformVersion());
         }
 
+        if (! template.getLaunchType().equals("Default Capacity Provider")) {
```
Comment (Chris9224) [inline src/main/java/com/cloudbees/jenkins/plugins/amazonecs/ECSService.java:423]: Have "Default Capacity Provider" as a constant perhaps?

File: src/main/java/com/cloudbees/jenkins/plugins/amazonecs/ECSTaskTemplate.java [hunk 2]
```diff
@@ -980,6 +980,7 @@ public ListBoxModel doFillLaunchTypeItems() {
             for (LaunchType launchType: LaunchType.values()) {
                 options.add(launchType.toString());
             }
+            options.add("Default Capacity Provider");
```
Comment (Chris9224) [inline src/main/java/com/cloudbees/jenkins/plugins/amazonecs/ECSTaskTemplate.java:983]: Here we can reuse a constant for "Default Capacity Provider" if we have one

**Comment by chathsuom** [comment]
Inspired by this change, added the same functionality of this + fargate spot capacity provider supports in this #211

**Comment by tdspencer3** [comment]
This PR really needs to be merged in. We are actively using this code in our production environment (going on 2 months now). But it means we had to build it and side-load it because the code has not been released in this plugin yet.

**Pull request closed without merging**